### PR TITLE
tos consent: show blocking modal for existing users that haven't consented yet

### PR DESCRIPTION
**Pull request by limitedmage**
For existing Sourcegraph Server (self-hosted) users, we need to block account access until they consent to the terms of service. This also applies to new users created via Oauth or via admin invite.

Part of #28716

image

### CODE REVIEW

File: client/web/src/auth/TosConsentModal.module.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+.container {
+    display: flex;
+    height: 100%;
+    flex-direction: column;
+    padding: 1.5rem;
+}
```
Comment (kopancek) [inline client/web/src/auth/TosConsentModal.module.scss:6]: Don't we already have some standard layouts for modals that we could leverage here instead? Not sure, asking :)

File: client/web/src/Layout.tsx [hunk 2]
```diff
@@ -218,11 +219,27 @@ export const Layout: React.FunctionComponent<LayoutProps> = props => {
...
     if (props.location.pathname !== '/' && props.location.pathname.endsWith('/')) {
         return <Redirect to={{ ...props.location, pathname: props.location.pathname.slice(0, -1) }} />
     }
 
+    // If a user has not accepted the Terms of Service yet, show the modal to force them to accept
+    // before continuing to use Sourcegraph. This is only done on self-hosted Sourcegraph Server;
+    // cloud users are all considered to have accepted regarless of the value of `tosAccepted`.
+    if (!props.isSourcegraphDotCom && !tosAccepted) {
+        return <TosConsentModal afterTosAccepted={afterTosAccepted} />
+    }
```
Comment (kopancek) [inline client/web/src/Layout.tsx:241]: Not sure if you checked with @ryphil or legal, but we might want to apply the TOS Modal to Cloud as well...

**Review by kopancek — APPROVED** [review]
Looks good to me.

File: client/web/src/auth/TosConsentModal.module.scss (hunk 1 above)
Comment (limitedmage) [reply client/web/src/auth/TosConsentModal.module.scss:6]: This is not really a traditional modal where it's a popup window with a translucent background where you can still see the page behind this. This is a full-page modal and no other content will be rendered on the page when this is visible.

File: client/web/src/Layout.tsx (hunk 2 above)
Comment (limitedmage) [reply client/web/src/Layout.tsx:241]: I did check and it's not needed on Cloud. On Cloud we already show the ToS links on every account creation.

**Review by marisak — COMMENTED** [review]
Looks great! Non-blocking nit: I don't think "Sign Out" needs to be capitalized (just a link) but I'm open to being wrong about that. Thanks so much for your hard work on this!

**Pull request merged**